test/DurableTaskEndToEndTests.cs

```diff
@@ -58,6 +58,36 @@ public DurableTaskEndToEndTests(ITestOutputHelper output)
             }
         }
 
+        /// <summary>
+        /// End-to-end test which validates a simple orchestrator function which doesn't call any activity functions.
+        /// </summary>
+        [Fact]
+        public async Task HelloWorldOrchestration_StartNewAsync_Polling()

```

Comment:
If you agree that we should remove the `waitUntilOrchestrationStarts`, then I think we don't need a new unit test. Instead, we can simply add your Stopwatch logic and Elapsed validation to the existing unit test.